GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -38,5 +39,27 @@ public override string GetDataRootForGVFSComponent(string componentName)
         {
             return new MacFileBasedLock(fileSystem, tracer, lockPath);
         }
+
+        public class MacPlatformConstants : GVFSPlatformConstants
+        {
+            public MacPlatformConstants() : base(executableExtension: string.Empty, installerExtension: ".dmg")

```

Comment:
I don't like how we are now setting the platform values via the constructor and with overrides of properties.  Also will we need to have these for POSIX so why are they on Mac?   Also won't the other executable names that are in `GVFSPlatformConstants` need to be overridden if `GVFSExecutableName` needs to be for case only?

I was also wondering if we should have interfaces like `IGVFSPaths` and `IExecutables` for the platforms to kind of break this up.

cc:  @wilbaker for his thoughts as well.

Reply:
> I don't like how we are now setting the platform values via the constructor and with overrides of properties

I agree, until we do a larger refactoring of this code could we stick with the current pattern (no overrides)?

Reply:
Discussed with @kewillford. Made `GVFSPlatformConstants` an abstract class. Removed setting platform constants using Constructor. Updated Platform specific constants to implement getters for executable binary and installer extensions.